lib/thrift/binary/framed/client.ex

```diff
@@ -125,6 +125,10 @@ defmodule Thrift.Binary.Framed.Client do
         Logger.error("Connection closed")
         {:stop, error, nil}
 
+      {:error, :timeout} = error ->
+        Logger.error("Connection timed out (#{timeout}ms)")
```

Comment:
The `timeout` is the default timeout, which won't be `1` in the test below. We could do something like

```elixir
assert ExUnit.CaptureLog.capture_log(fn ->
  assert {:error, :timeout} = Client.friend_ids_of_with_options(client, 12_914, [tcp_opts: [timeout: 1]])
  assert_receive {:EXIT, ^client, {:error, :timeout}}
 end) =~ "1ms"
```

Reply:
Ah, you're right, and because the recv timeout option can be set on a per-call basis, that value isn't available in `disconnect/2` unless we included it as a new term in the error tuple (e.g. `{:error, :timeout, 1}`). Is that a standard approach?

Alternatively, I could log from `handle_call/2` or not log the timeout value at all.

Reply:
I think that logging the timeout length is very useful. Usually the value is passed to disconnect callback and logged there, so any method similar to you suggested (append to the tuple) is good.